graph.go

```diff
@@ -55,22 +57,33 @@ var graphCmd GraphCmd
 var allowErrorsInGoGet = true
 
 func (c *GraphCmd) Execute(args []string) error {
-	var unit *unit.SourceUnit
-	if err := json.NewDecoder(os.Stdin).Decode(&unit); err != nil {
+	inputBytes, err := ioutil.ReadAll(os.Stdin)
+	if err != nil {
 		return err
 	}
+	var units unit.SourceUnits
+	if err := json.NewDecoder(bytes.NewReader(inputBytes)).Decode(&units); err != nil {
+		// Legacy API: try parsing input as a single source unit
+		var unit *unit.SourceUnit
+		if err := json.NewDecoder(bytes.NewReader(inputBytes)).Decode(&unit); err != nil {
+			return err
+		}
+		units = append(units, unit)
```

Comment:
I would say that `units = []*unit.SourceUnit{unit}` is more _appropriate_, since the intention is to clear whatever might be in `units` (in practice, likely nothing will be there) and assign the value of `unit` to it. It's because you're effectively doing a fallback to parsing as legacy API here, so it's not an additive operation, it's a replacement operation.